keysync/server.go

```diff
@@ -233,26 +233,22 @@ func (ks *KeySyncServer) syncSecretsToLocalKeys(secList *corev1.SecretList, skh
 					logrus.Errorf("Unable to write file %s: %v", path, err)
 					continue
 				}
-			}
 
-			// Permission and ownership corrigation if needed
-			fileInfo, err := os.Stat(path)
-			if err != nil {
-				logrus.Errorf("Unable to get file %s: %v", path, err)
-			}
-			if fileInfo.Mode() != ks.keyFilePermissions {
-				err = os.Chmod(path, ks.keyFilePermissions)
+				// Permission corrigation might be needed as WriteFile does not
+				// guarantee the specified permissions due to umask
+				err = filePermissionCorrigation(path, ks.keyFilePermissions)
```

Comment:
Hmm , how about making this ks.WriteFile(path, data), and then it can get all the file permission configurations and do everything within the block:
```
if !fileExists(path) {
    err := ks.writeKeyFile(path, data)
    if err != nil {
        logrus.Errorf("Unable to write file %s: %v", path, err)
        continue
	}

}
```

Reply:
created the `writeKeyFile` func and performing everything in that function